sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -534,71 +535,55 @@ Mono<AttestationResponse<AttestationResult>> attestSgxEnclaveWithResponse(Attest
         return withContext(context -> this.attestTpmWithResponse(request, context));
     }
 
-    /** Performs TPM attestation.
+    /**
+     * @brief Performs TPM attestation.
+     *
+     * The TPM attestation protocol is defined <a href='https://docs.microsoft.com/azure/attestation/virtualization-based-security-protocol'>here.</a></p>
      *
-     * Processes attestation evidence from a VBS enclave, producing an attestation result.
-     * <p>The TPM attestation protocol is defined <a href='https://docs.microsoft.com/azure/attestation/virtualization-based-security-protocol'>here.</a></p>
-     * <p>Unlike OpenEnclave reports and SGX enclave quotes, TPM attestation is implemented using JSON encoded
-     * strings. </p><p>The client formats a string serialized JSON request to the service, which responds with a
-     * JSON response. The serialized JSON object exchange continues until the service responds with a JSON string
-     * with a property named {@code "report"}, whose value will be an attestation result token.</p>
-     * <p><strong>Perform the first leg of a TPM attestation operation</strong></p>
-     * <!-- src_embed com.azure.security.attestation.AttestationAsyncClient.attestTpm -->
+     * <p><strong>Attest using TPM.</strong></p>
      * <pre>
-     * &#47;&#47; The initial payload for TPM attestation is a JSON object with a property named &quot;payload&quot;,
-     * &#47;&#47; containing an object with a property named &quot;type&quot; whose value is &quot;aikcert&quot;.
+     * Mono&lt;TpmAttestationResult&gt; result = client.attestTpm&#40;BinaryData.fromString&#40;attestInitialPayload&#41;&#41;;
      *
-     * String attestInitialPayload = &quot;&#123;&#92;&quot;payload&#92;&quot;: &#123; &#92;&quot;type&#92;&quot;: &#92;&quot;aikcert&#92;&quot; &#125; &#125;&quot;;
-     * Mono&lt;String&gt; tpmResponse = client.attestTpm&#40;attestInitialPayload&#41;;
      * </pre>
-     * <!-- end com.azure.security.attestation.AttestationAsyncClient.attestTpm -->
-     *
+     * 
      * @param request Attestation request for Trusted Platform Module (TPM) attestation.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws HttpResponseException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return attestation response for Trusted Platform Module (TPM) attestation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException    thrown if the request is rejected by server.
+     * @throws RuntimeException         all other wrapped checked exceptions if the request fails to be sent.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<String> attestTpm(String request) {
+    public Mono<TpmAttestationResult> attestTpm(BinaryData request) {
```

Comment:
> https://github.com/Azure/azure-sdk-for-java/pull/33739/files#r1117809151
> If the is BinaryData, the output should also be BinaryData. If the Output is TpmAttestationResult, the input should be TpmAttestationRequest.
>
> Be consistent with both of these.
> @LarryOsterman 

I believe we will not add a TpmAttestationRequest class since there is currently no AttestationRequest class either and so that we will take BinaryData exactly like AttestSgxEnclave() does.

Reply:
Each AttestionType has it own set of inputs and outputs, and you should not use one as a model for the other, 

In general, it is considered a *poor* design pattern to use content sniffing to distinguish between (for instance) a JSON  object and a binary format.

When you send JSON data masquerading as BinaryData, it depends on the server sniffing the content to see if it is JSON data. If you have a TpmAttestationRequest, you have the ability to annotate that the BinaryData is JSON or BinaryData - this is consistent with how RuntimeData and InitTimeData are both expressed - they are structures which have a discriminator which expresses how the data field should be interpreted.